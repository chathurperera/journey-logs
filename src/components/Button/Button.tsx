/* eslint-disable @typescript-eslint/no-use-before-define */
import { ButtonProps as RNEButtonProps } from '@rneui/themed';
import React from 'react';
import { StyleSheet } from 'react-native';
import { Button as RNEButton } from 'react-native-elements';

import { Color } from '@jl/constants';

interface ButtonProps extends RNEButtonProps {
  backgroundColor?: string;
  textColor?: string;
}

export function Button({
  title,
  backgroundColor = Color.Neutral.black,
  textColor = Color.Neutral.white,
  onPress,
  loading,
  disabled,
}: ButtonProps) {
  return (
    <RNEButton
      onPress={onPress}
      disabled={disabled}
      title={title}
      buttonStyle={[{ backgroundColor: backgroundColor }, styles.buttonStyles]}
      titleStyle={{ color: textColor }}
      loading={loading}
    />
  );
}

const styles = StyleSheet.create({
  buttonStyles: {
    borderRadius: 5,
    paddingVertical: 11,
  },
});
