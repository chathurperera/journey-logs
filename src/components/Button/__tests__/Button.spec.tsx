import { render } from '@testing-library/react-native';
import * as React from 'react';

import { Button } from '../Button';

describe('Button component', () => {
  const testID = 'ButtonTestID';
  const getRenderedButton = () => render(<Button testID={testID} />);

  describe('should render correctly', () => {
    const rendered = getRenderedButton();
    const renderedTree = rendered.toJSON();
    expect(renderedTree).toMatchSnapshot();
  });

  it('should find the button via testID', () => {
    const { getByTestId } = getRenderedButton();
    const foundButton = getByTestId(testID);

    expect(foundButton).toBeTruthy();
  });
});
