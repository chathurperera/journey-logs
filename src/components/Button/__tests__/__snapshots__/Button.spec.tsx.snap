// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    [
      {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": false,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="ButtonTestID"
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "borderColor": "#2089dc",
          "borderRadius": 10,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
          "paddingBottom": 12,
          "paddingTop": 12,
        }
      }
    />
  </View>
</View>
`;
